packages/typegpu/src/core/root/rootTypes.ts

```diff
@@ -592,6 +592,10 @@ export interface TgpuRoot extends Unwrapper {
 
 export interface ExperimentalTgpuRoot extends TgpuRoot, WithBinding {
   readonly nameRegistry: NameRegistry;
+  /** Experimental: custom shader generator used for function bodies during resolution */
+  readonly unstable_shaderGenerator?:
+    | import('../../types').ShaderGenerator
```

Comment:
Why import the type like this? I feel like we could just import it top-level

```ts
import type { ShaderGenerator } from '../../types.ts';
```

```suggestion
    | ShaderGenerator
```